manifests/config.pp

```diff
@@ -120,5 +120,24 @@
         changes => template('foreman_proxy/sudo_augeas.erb'),
       }
     }
+  } else {
+    # The puppet-agent (puppet 4 AIO package) doesn't create a puppet user and group
+    # but the foreman proxy still needs to be able to read the agent's private key
+    if $foreman_proxy::manage_puppet_group {
+      if !defined(Group[$foreman_proxy::puppet_group]) {
+        group { $foreman_proxy::puppet_group:
```

Comment:
If you like.  I was probably just taking advantage of the existing `if` statement.  Autorequires *should* mean ordering is OK, but it'd perhaps be clearer if I just moved it.